File: simpleplayer/Python/ai.py
import random
import pprint

class AI:
	def einsatz(self):
		"""Diese Funkion wird jede Runde aufgerufen, und gibt die Punkte zurück, die gesetzt werden sollen."""
		einsatz = 5
		print("setze:", einsatz)
		return einsatz

	def process(self, ownWonChips, enemyWonChips, ownChips, enemyChips):
		"""Diese Funkion wird nach jedem Schritt aufgerufen und ist dafür da, Daten zu verarbeiten.
		Es werden ownWonChips   (gesammten Chips, die die eigene KI bekommen hat),
		          enemyWonChips (gesammten Chips, die der gegner bekommen hat),
		          ownChips      (Chips, die man diese Runde gesetzt hat),
		          enemyChips    (Chips, die der Gegner diese Runde gesetzt hat) übergeben."""
		pprint.pprint({
			"ownWonChips": ownWonChips,
			"enemyWonChips": enemyWonChips,
			"ownChips": ownChips,
			"enemyChips": enemyChips
		})

File: simpleplayer/Go/src/ai/ai.go
package ai

import "fmt"

// RedirectOutput legt fest, ob Debugausgaben weitergeleited werden sollen. (ca. +0.1ms pro Schritt)
// Diese Option auf 'false' zu Setzen macht nur bei eine Quali-KI Sinn.
const RedirectOutput = false

// ProcessData legt fest, ob Informationen verarbeitet werden sollen. (ca. +0.1ms pro Schritt)
// Diese Option auf 'false' zu Setzen macht nur bei eine Quali-KI Sinn.
const ProcessData = false

// Init wird vor dem ersten Einsatz ausgeführt.
func Init() {
	fmt.Println("Hallo, Welt")
}

// Einsatz gibt den Einsatz der KI zurück
func Einsatz() int {
	fmt.Println("Setze 5")
	return 5
}

// Process verarbeitet Daten des letzten Schrittes
func Process(ownWonChips, enemyWonChips, ownChips, enemyChips int) {
	fmt.Println(ownWonChips, enemyWonChips, ownChips, enemyChips)
}
